Userland/Libraries/LibJS/Bytecode/CommonImplementations.cpp

```diff
@@ -68,23 +104,43 @@ ThrowCompletionOr<Value> get_by_id(VM& vm, DeprecatedFlyString const& property,
 
 ThrowCompletionOr<Value> get_by_value(VM& vm, Value base_value, Value property_key_value)
 {
-    auto object = TRY(base_object_for_get(vm, base_value));
-
     // OPTIMIZATION: Fast path for simple Int32 indexes in array-like objects.
-    if (property_key_value.is_int32() && property_key_value.as_i32() >= 0) {
+    if (base_value.is_object() && property_key_value.is_int32() && property_key_value.as_i32() >= 0) {
+        auto& object = base_value.as_object();
         auto index = static_cast<u32>(property_key_value.as_i32());
 
         // For "non-typed arrays":
-        if (!object->may_interfere_with_indexed_property_access()
-            && object->indexed_properties().has_index(index)) {
-            auto value = object->indexed_properties().get(index)->value;
+        if (!object.may_interfere_with_indexed_property_access()
+            && object.indexed_properties().has_index(index)) {
+            auto value = object.indexed_properties().get(index)->value;
             if (!value.is_accessor())
                 return value;
         }
 
         // For typed arrays:
-        if (object->is_typed_array()) {
-            auto& typed_array = static_cast<TypedArrayBase&>(*object);
+        if (object.is_typed_array()) {
+            auto& typed_array = static_cast<TypedArrayBase&>(object);
+
+            if (!typed_array.viewed_array_buffer()->is_detached() && index < typed_array.array_length()) {
+                switch (typed_array.kind()) {
+                case TypedArrayBase::Kind::Uint8Array:
```

Comment:
Can't you use the macro here?

Reply:
I’ve only implemented this path for some of the array kinda, but sure if we figure out a nice generic way to do all kinds then we can use the macro :)

Reply:
Sure, my bad!